src/client.c

```diff
@@ -4454,6 +4454,53 @@ void CheckTeamStatus()
 	}
 }
 
+void SendSpecInfo()
+{
+	gedict_t *t, *p;
+	int cl;
+	char *tracking, *nick;
+
+	static double lastupdate = 0;
+
+	if (g_globalvars.time - lastupdate < 2)
+	{
+		return;
+	}
+
+	lastupdate = g_globalvars.time;
+
+	for (t = world; (t = find_spc(t));)
+	{
+		if (t->ct != ctSpec)
+		{
+			continue;
+		}
+
+		cl = NUM_FOR_EDICT(t) - 1;
+		tracking = TrackWhom(t);
```

Comment:
you should send number instead and guess player name in client, also, I doubt you should send spectator nick as well, just send his number